src/types.ts

```diff
@@ -0,0 +1,255 @@
+export type BlockHeight = number;
+
+export interface LakeConfig {
+  s3Endpoint?: string;
+  s3BucketName: string;
+  s3RegionName: string;
+  startBlockHeight: number;
+};
+
+export interface StreamerMessage {
+  block: Block;
+  shards: Shard[];
+};
+
+export interface Block {
+  author: string;
+  header: BlockHeader;
+  chunks: ChunkHeader[];
+};
+
+export type ValidatorProposal = {
+  accountId: string;
+  publicKey: string;
+  stake: string;
+  validatorStakeStructVersion: string;
+};
+
+export type ChallengeResult = {
+  accountId: string;
+  isDoubleSign: boolean;
+};
+
+export interface BlockHeader {
+  approvals: (string | null)[];
+  blockMerkleRoot: string;
+  blockOrdinal: number;
+  challengesResult: ChallengeResult[];
+  challengesRoot: string;
+  chunkHeadersRoot: string;
+  chunkMask: boolean[];
+  chunkReceiptsRoot: string;
+  chunkTxRoot: string;
+  chunksIncluded: number;
+  epochId: string;
+  epochSyncDataHash: string | null;
+  gasPrice: string;
+  hash: string;
+  height: number;
+  lastDsFinalBlock: string;
+  lastFinalBlock: string;
+  latestProtocolVersion: number;
+  nextBpHash: string;
+  nextEpochId: string;
+  outcomeRoot: string;
+  prevHash: string;
+  prevHeight: number;
+  prevStateRoot: string;
+  randomValue: string;
+  rentPaid: string;
+  signature: string;
+  timestamp: number;
+  timestampNanosec: string;
+  totalSupply: string;
+  validatorProposals: [];
+  validatorReward: string;
+};
+
+export interface Shard {
+  shard_id: number;
+  chunk?: Chunk;
+  receiptExecutionOutcomes: ExecutionOutcomeWithReceipt[];
+  stateChanges: StateChange[];
+};
+
+export interface ChunkHeader {
+  balanceBurnt: number;
+  chunkHash: string;
+  encodedLength: number;
+  encodedMerkleRoot: string;
+  gasLimit: number;
+  gasUsed: number;
+  heightCreated: number;
+  heightIncluded: number;
+  outcomeRoot: string;
+  outgoingReceiptsRoot: string;
+  prevBlockHash: string;
+  prevStateRoot: string;
+  rentPaid: string;
+  shardId: number;
+  signature: string;
+  txRoot: string;
+  validatorProposals: ValidatorProposal[];
+  validatorReward: string;
+};
+
+export interface Chunk {
+  author: string;
+  header: ChunkHeader;
+  receipts: Receipt[],
+  transactions: IndexerTransactionWithOutcome[];
+}
+
+export type ReceiptEnum =
+  | {
+      Action: {
+        actions: Action[];
+        gasPrice: string;
+        inputDataIds: string[];
+        outputDataReceivers: string[];
+        signerId: string;
+        signerPublicKey: string;
+      };
+    }
+    | {
+      Data: {
+        data: string;
+        dataId: string;
+      };
+    };
+
+export type Receipt = {
+  predecessorId: string;
+  receipt: ReceiptEnum;
+  receiptId: string;
+  receiverId: string;
+}
+
+export type DeployContractAction = {
+  DeployContract: {
+    code: string;
+  };
+};
+
+export type FunctionCallAction= {
```

Comment:
There is a typo.
I think you can install a basic setup of `prettier` to fix this kind of typos.